### PR TITLE
Add PostgreSQL as DB Option

### PR DESCRIPTION
**Pull request by nickmurray47**
Testing steps:
- `go mod vendor`
- update `config.yaml` to have `postgres` as database type
- from `fasten-onprem/backend/pkg/models/database` run `go run generate.go` and verify that all the database models have swapped out `datetime` for `timestamptz`

### CODE REVIEW

**Comment by nickmurray47** [comment]
to-do:
- create common Gorm file for sqlite/postgres versions to inherit from
- create factory file for generating database/postrgres/ folder

**Comment by AnalogJ** [comment]
Hey Nick, I'm worried that this PR might end up requiring a long-lived branch if we start making the "common" repository changes and adding the Postgres repository stuff now
Can we move that to another PR, and merge just the Factory pattern + model changes?

**Comment by nickmurray47** [comment]
> I'm worried that this PR might end up requiring a long-lived branch if we start making the "common" repository changes and adding the Postgres repository stuff now
Can we move that to another PR, and merge just the Factory pattern + model changes?

Yes, I was actually going to stop here and ping you for a review. Do you still want to move the "common" + postgres changes out? I just re-tested with SQlite as default and everything appeared operational.

**Comment by AnalogJ** [comment]
Honestly I’ll just worried that there are changes in the main branch that
may not be reflected in yours. Do you have time for a chat on Monday
morning?

**Comment by nickmurray47** [comment]
Yeah, can chat Monday morning!

**Comment by AnalogJ** [comment]
I think the merge commit is fine.

File: backend/pkg/models/database/generate.go [hunk 1]
```diff
@@ -47,6 +51,22 @@ PLEASE DO NOT EDIT BY HAND
 `, "\n"), "\n")
 
 func main() {
+	// Read config file for database type
+	appconfig, err := config.Create()
```
Comment (AnalogJ) [inline backend/pkg/models/database/generate.go:55]: Compile time vs runtime. 
The user's configuration file will not be present during compilation time, so the database type choice has not been made yet. 
We need to generate both Sqlite & Postgres compatible structs (in different directories).
The generated `utils.NewFhirResourceModelByType` fn should take in the DatabaseRepositoryType as a parameter, and conditionally point to a different set of generated structs.

**Pull request merged**